pkg/volume/util/util.go

```diff
@@ -164,20 +164,20 @@ func GetClassForVolume(kubeClient clientset.Interface, pv *v1.PersistentVolume)
 
 // CheckNodeAffinity looks at the PV node affinity, and checks if the node has the same corresponding labels
 // This ensures that we don't mount a volume that doesn't belong to this node
-func CheckNodeAffinity(pv *v1.PersistentVolume, nodeLabels map[string]string) error {
-	return checkVolumeNodeAffinity(pv, nodeLabels)
-}
-
-func checkVolumeNodeAffinity(pv *v1.PersistentVolume, nodeLabels map[string]string) error {
+func CheckNodeAffinity(pv *v1.PersistentVolume, node *v1.Node) error {
 	if pv.Spec.NodeAffinity == nil {
 		return nil
 	}
 
 	if pv.Spec.NodeAffinity.Required != nil {
 		terms := pv.Spec.NodeAffinity.Required.NodeSelectorTerms
 		klog.V(10).Infof("Match for Required node selector terms %+v", terms)
-		if !v1helper.MatchNodeSelectorTerms(terms, labels.Set(nodeLabels), nil) {
-			return fmt.Errorf("no matching NodeSelectorTerms")
+		if matches, err := corev1.MatchNodeSelectorTerms(node, terms); !matches {
+			errString := ""
+			if err != nil {
+				errString = fmt.Sprintf("(error: %+v", err)
+			}
+			return fmt.Errorf("no matching NodeSelectorTerms %s", errString)
```

Comment:
```suggestion
		if matches, err := corev1.MatchNodeSelectorTerms(node, terms); err != nil {
			return err
		} else if !matches {
			return fmt.Errorf("no matching NodeSelectorTerms %s", errString)
		}	
```

But kind of weird that this function doesn't return (bool, error)

Reply:
Updated